### PR TITLE
Switch to community Picker

### PR DESCRIPTION
**Pull request by abumostafa**
Fixes: https://github.com/lawnstarter/react-native-picker-select/issues/301

image

### CODE REVIEW

**Comment by lfkwtz** [comment]
1. there should be lockfile changes
2. heads up that this is a significant breaking change for this library and won't be merged in anytime soon. we'll have to come up with a reasonable migration path.

**Comment by abumostafa** [comment]
> 1. there should be lockfile changes
> 2. heads up that this is a significant breaking change for this library and won't be merged in anytime soon. we'll have to come up with a reasonable migration path.

Thanks for your reply.

1. Sorry I'm using `npm` and did not notice `yarn.lock` file. but will update accordingly.
2. Does it make sense to have 2 releases to support `RN < 0.62` and `RN >= 0.62`??

**Comment by lfkwtz** [comment]
The community Picker could’ve been used long before 0.62 - the deprecation warning is just now bringing it to light.

The larger concern is the move away from a pure JavaScript implementation and how that will affect Expo users. 

Since we’re only at a deprecation warning stage right now, and only for users running the latest RN version, the impact of this is low enough for now that’d I’d recommend disabling the warning and moving on.

I’d rather not have to maintain separate versions. The likely path forward will be fixing other bugs / issues that result in “smaller” breaking changes before we reach a cut off point and force the use of the community Picker.

**Comment by abumostafa** [comment]
> The community Picker could’ve been used long before 0.62 - the deprecation warning is just now bringing it to light.
> 
> The larger concern is the move away from a pure JavaScript implementation and how that will affect Expo users.
> 
> Since we’re only at a deprecation warning stage right now, and only for users running the latest RN version, the impact of this is low enough for now that’d I’d recommend disabling the warning and moving on.
> 
> I’d rather not have to maintain separate versions. The likely path forward will be fixing other bugs / issues that result in “smaller” breaking changes before we reach a cut off point and force the use of the community Picker.

I agree with your concerns and it does not make much sense to maintain 2 versions. However sooner or later the `Picker` will be removed from `RN` core. 

IMHO we can check the impact on Expo users. If there is nothing to worry about I would suggest migrating to community picker. otherwise, It makes sense to delay the migration until it's safer for everyone. 

You could share your opinion and I'm happy to help either way.

**Comment by lfkwtz** [comment]
Anything you (or anyone) can find out about the impact to Expo users and what the plan forward for Expo is in regards to this deprecation would be immensely appreciated

**Comment by lfkwtz** [comment]
https://github.com/react-native-community/discussions-and-proposals/issues/120
could be the answer here

File: src/index.js [hunk 1]
```diff
@@ -12,6 +11,7 @@ import {
 import PropTypes from 'prop-types';
 import isEqual from 'lodash.isequal';
 import { defaultStyles } from './styles';
+import { Picker } from "@react-native-community/picker";
```
Comment (mattmcdonald-uk) [inline src/index.js:14]:
```suggestion
import { Picker } from '@react-native-community/picker';
```

**Comment by Naturalclar** [comment]
@lfkwtz about covering expo users, expo supports files like `xxx.expo.js` that will be read on expo projects.
It may not be a clean solution, but if there is an `index.expo.js` that imports Picker from react native, then expo user will not have to experience the breaking change.

```
// index.expo.js
import {Picker} from 'react-native'
```

```
// index.js
import {Picker} from '@react-native-community/picker'
```

**Comment by lfkwtz** [comment]
> @lfkwtz about covering expo users, expo supports files like `xxx.expo.js` that will be read on expo projects.
> 
> It may not be a clean solution, but if there is an `index.expo.js` that imports Picker from react native, then expo user will not have to experience the breaking change.
> 
> 
> 
> ```
> 
> // index.expo.js
> 
> import {Picker} from 'react-native'
> 
> ```
> 
> 
> 
> ```
> 
> // index.js
> 
> import {Picker} from '@react-native-community/picker'
> 
> ```

Good to know. Can you link to docs for that?

**Comment by mattmcdonald-uk** [comment]
> Good to know. Can you link to docs for that?

https://docs.expo.io/bare/using-expo-client/#use-the-expojsjsontstsx-extension-to-provide-expo

**Comment by linzera** [comment]
Nice work guys!

File: package.json [hunk 2]
```diff
@@ -29,6 +29,9 @@
     },
     "devDependencies": {
         "babel-jest": "^23.6.0",
+        "babel-plugin-add-module-exports": "^1.0.2",
```
Comment (mir4a) [inline package.json:33]: Do you need these babel plugins and preset since you ignore transformation of Picker for Jest?

**Comment by boerdjamin** [comment]
Stupid question incoming: How do I get the changes in my project? There is now version bump yet as far as I can see

**Comment by ChristophDietrich** [comment]
@boerdjamin 

You can go by directly linking it in the package.json with

"react-native-picker-select": "lawnstarter/react-native-picker-select"

**Comment by chenop** [comment]
any updates? can this be merged?

**Pull request closed without merging**